// tencentcloud-sdk-nodejs/es/services/tdmysql/v20211122/tdmysql_client.js
import { AbstractClient } from "../../../common/abstract_client";
export class Client extends AbstractClient {
    constructor(clientConfig) {
        super("tdmysql.tencentcloudapi.com", "2021-11-22", clientConfig);
    }
    async DescribeDatabaseObjects(req, cb) {
        return this.request("DescribeDatabaseObjects", req, cb);
    }
    async DescribeDBParameters(req, cb) {
        return this.request("DescribeDBParameters", req, cb);
    }
    async DescribeDBSecurityGroups(req, cb) {
        return this.request("DescribeDBSecurityGroups", req, cb);
    }
    async CancelIsolateDBInstances(req, cb) {
        return this.request("CancelIsolateDBInstances", req, cb);
    }
    async DescribeDBSAvailableRecoveryTime(req, cb) {
        return this.request("DescribeDBSAvailableRecoveryTime", req, cb);
    }
    async ModifyDBSBackupSetComment(req, cb) {
        return this.request("ModifyDBSBackupSetComment", req, cb);
    }
    async DeleteDBSBackupSets(req, cb) {
        return this.request("DeleteDBSBackupSets", req, cb);
    }
    async ModifyDBSBackupPolicy(req, cb) {
        return this.request("ModifyDBSBackupPolicy", req, cb);
    }
    async DestroyInstances(req, cb) {
        return this.request("DestroyInstances", req, cb);
    }
    async ModifyAutoRenewFlag(req, cb) {
        return this.request("ModifyAutoRenewFlag", req, cb);
    }
    async CreateDBSBackup(req, cb) {
        return this.request("CreateDBSBackup", req, cb);
    }
    async ModifyInstanceName(req, cb) {
        return this.request("ModifyInstanceName", req, cb);
    }
    async ModifyDBParameters(req, cb) {
        return this.request("ModifyDBParameters", req, cb);
    }
    async DescribeDBSCloneInstances(req, cb) {
        return this.request("DescribeDBSCloneInstances", req, cb);
    }
    async IsolateDBInstance(req, cb) {
        return this.request("IsolateDBInstance", req, cb);
    }
    async ModifyDBInstanceSecurityGroups(req, cb) {
        return this.request("ModifyDBInstanceSecurityGroups", req, cb);
    }
    async DescribeFlow(req, cb) {
        return this.request("DescribeFlow", req, cb);
    }
}
