// tencentcloud-sdk-nodejs/es/services/cls/v20201016/cls_client.js
import { AbstractClient } from "../../../common/abstract_client";
export class Client extends AbstractClient {
    constructor(clientConfig) {
        super("cls.tencentcloudapi.com", "2020-10-16", clientConfig);
    }
    async ModifyKafkaConsumer(req, cb) {
        return this.request("ModifyKafkaConsumer", req, cb);
    }
    async CreateKafkaRecharge(req, cb) {
        return this.request("CreateKafkaRecharge", req, cb);
    }
    async DescribeExports(req, cb) {
        return this.request("DescribeExports", req, cb);
    }
    async ModifyNetworkApplication(req, cb) {
        return this.request("ModifyNetworkApplication", req, cb);
    }
    async ModifyHostMetricConfig(req, cb) {
        return this.request("ModifyHostMetricConfig", req, cb);
    }
    async GetAlarmLog(req, cb) {
        return this.request("GetAlarmLog", req, cb);
    }
    async CreateConsumerGroup(req, cb) {
        return this.request("CreateConsumerGroup", req, cb);
    }
    async CreateIndex(req, cb) {
        return this.request("CreateIndex", req, cb);
    }
    async QueryMetric(req, cb) {
        return this.request("QueryMetric", req, cb);
    }
    async ModifyDlcDeliver(req, cb) {
        return this.request("ModifyDlcDeliver", req, cb);
    }
    async DeleteConsumerGroup(req, cb) {
        return this.request("DeleteConsumerGroup", req, cb);
    }
    async DescribeConfigs(req, cb) {
        return this.request("DescribeConfigs", req, cb);
    }
    async CreateDeliverCloudFunction(req, cb) {
        return this.request("CreateDeliverCloudFunction", req, cb);
    }
    async DeleteLogset(req, cb) {
        return this.request("DeleteLogset", req, cb);
    }
    async DeleteConfigFromMachineGroup(req, cb) {
        return this.request("DeleteConfigFromMachineGroup", req, cb);
    }
    async DeleteDataTransform(req, cb) {
        return this.request("DeleteDataTransform", req, cb);
    }
    async CreateLogset(req, cb) {
        return this.request("CreateLogset", req, cb);
    }
    async DescribeNetworkApplications(req, cb) {
        return this.request("DescribeNetworkApplications", req, cb);
    }
    async DescribeEsRecharges(req, cb) {
        return this.request("DescribeEsRecharges", req, cb);
    }
    async CreateTopic(req, cb) {
        return this.request("CreateTopic", req, cb);
    }
    async DescribeCloudProductLogTasks(req, cb) {
        return this.request("DescribeCloudProductLogTasks", req, cb);
    }
    async DescribeHostMetricConfigs(req, cb) {
        return this.request("DescribeHostMetricConfigs", req, cb);
    }
    async CreateMetricSubscribe(req, cb) {
        return this.request("CreateMetricSubscribe", req, cb);
    }
    async DescribeKafkaConsumerPreview(req, cb) {
        return this.request("DescribeKafkaConsumerPreview", req, cb);
    }
    async DescribeClusterMetricConfigs(req, cb) {
        return this.request("DescribeClusterMetricConfigs", req, cb);
    }
    async DescribeAlertRecordHistory(req, cb) {
        return this.request("DescribeAlertRecordHistory", req, cb);
    }
    async DeleteKafkaRecharge(req, cb) {
        return this.request("DeleteKafkaRecharge", req, cb);
    }
    async CreateConfig(req, cb) {
        return this.request("CreateConfig", req, cb);
    }
    async DescribeDlcDelivers(req, cb) {
        return this.request("DescribeDlcDelivers", req, cb);
    }
    async DescribeTopics(req, cb) {
        return this.request("DescribeTopics", req, cb);
    }
    async CancelRebuildIndexTask(req, cb) {
        return this.request("CancelRebuildIndexTask", req, cb);
    }
    async EstimateRebuildIndexTask(req, cb) {
        return this.request("EstimateRebuildIndexTask", req, cb);
    }
    async DescribeSplunkPreview(req, cb) {
        return this.request("DescribeSplunkPreview", req, cb);
    }
    async CreateCloudProductLogCollection(req, cb) {
        return this.request("CreateCloudProductLogCollection", req, cb);
    }
    async CreateCosRecharge(req, cb) {
        return this.request("CreateCosRecharge", req, cb);
    }
    async MergePartition(req, cb) {
        return this.request("MergePartition", req, cb);
    }
    async DeleteHostMetricConfig(req, cb) {
        return this.request("DeleteHostMetricConfig", req, cb);
    }
    async DeleteScheduledSql(req, cb) {
        return this.request("DeleteScheduledSql", req, cb);
    }
    async CreateExport(req, cb) {
        return this.request("CreateExport", req, cb);
    }
    async DeleteSplunkDeliver(req, cb) {
        return this.request("DeleteSplunkDeliver", req, cb);
    }
    async DescribeClusterBaseMetricConfigs(req, cb) {
        return this.request("DescribeClusterBaseMetricConfigs", req, cb);
    }
    async DescribeConsumerOffsets(req, cb) {
        return this.request("DescribeConsumerOffsets", req, cb);
    }
    async DescribeNetworkApplicationDetail(req, cb) {
        return this.request("DescribeNetworkApplicationDetail", req, cb);
    }
    async DescribeTopicMetricConfigs(req, cb) {
        return this.request("DescribeTopicMetricConfigs", req, cb);
    }
    async ModifyMetricSubscribe(req, cb) {
        return this.request("ModifyMetricSubscribe", req, cb);
    }
    async CreateDlcDeliver(req, cb) {
        return this.request("CreateDlcDeliver", req, cb);
    }
    async ModifyIndex(req, cb) {
        return this.request("ModifyIndex", req, cb);
    }
    async ModifyKafkaRecharge(req, cb) {
        return this.request("ModifyKafkaRecharge", req, cb);
    }
    async ModifyCloudProductLogCollection(req, cb) {
        return this.request("ModifyCloudProductLogCollection", req, cb);
    }
    async CreateAlarmShield(req, cb) {
        return this.request("CreateAlarmShield", req, cb);
    }
    async DeleteMetricSubscribe(req, cb) {
        return this.request("DeleteMetricSubscribe", req, cb);
    }
    async CreateEsRecharge(req, cb) {
        return this.request("CreateEsRecharge", req, cb);
    }
    async ModifyEsRecharge(req, cb) {
        return this.request("ModifyEsRecharge", req, cb);
    }
    async DeleteDlcDeliver(req, cb) {
        return this.request("DeleteDlcDeliver", req, cb);
    }
    async DeleteCloudProductLogCollection(req, cb) {
        return this.request("DeleteCloudProductLogCollection", req, cb);
    }
    async DescribeEsRechargePreview(req, cb) {
        return this.request("DescribeEsRechargePreview", req, cb);
    }
    async CreateShipper(req, cb) {
        return this.request("CreateShipper", req, cb);
    }
    async CreateNoticeContent(req, cb) {
        return this.request("CreateNoticeContent", req, cb);
    }
    async DeleteConsumer(req, cb) {
        return this.request("DeleteConsumer", req, cb);
    }
    async DescribeConsumerPreview(req, cb) {
        return this.request("DescribeConsumerPreview", req, cb);
    }
    async DeleteMachineGroup(req, cb) {
        return this.request("DeleteMachineGroup", req, cb);
    }
    async DeleteEsRecharge(req, cb) {
        return this.request("DeleteEsRecharge", req, cb);
    }
    async SendConsumerHeartbeat(req, cb) {
        return this.request("SendConsumerHeartbeat", req, cb);
    }
    async DeleteCosRecharge(req, cb) {
        return this.request("DeleteCosRecharge", req, cb);
    }
    async DescribeKafkaConsumerGroupList(req, cb) {
        return this.request("DescribeKafkaConsumerGroupList", req, cb);
    }
    async ModifySplunkDeliver(req, cb) {
        return this.request("ModifySplunkDeliver", req, cb);
    }
    async DescribeKafkaRecharges(req, cb) {
        return this.request("DescribeKafkaRecharges", req, cb);
    }
    async DescribeAlarms(req, cb) {
        return this.request("DescribeAlarms", req, cb);
    }
    async DescribeCosRecharges(req, cb) {
        return this.request("DescribeCosRecharges", req, cb);
    }
    async ApplyConfigToMachineGroup(req, cb) {
        return this.request("ApplyConfigToMachineGroup", req, cb);
    }
    async CreateDashboard(req, cb) {
        return this.request("CreateDashboard", req, cb);
    }
    async DeleteAlarmShield(req, cb) {
        return this.request("DeleteAlarmShield", req, cb);
    }
    async DescribeMetricCorrectDimension(req, cb) {
        return this.request("DescribeMetricCorrectDimension", req, cb);
    }
    async DescribeLogHistogram(req, cb) {
        return this.request("DescribeLogHistogram", req, cb);
    }
    async ModifyAlarmNotice(req, cb) {
        return this.request("ModifyAlarmNotice", req, cb);
    }
    async OpenKafkaConsumer(req, cb) {
        return this.request("OpenKafkaConsumer", req, cb);
    }
    async DeleteDashboard(req, cb) {
        return this.request("DeleteDashboard", req, cb);
    }
    async ModifyShipper(req, cb) {
        return this.request("ModifyShipper", req, cb);
    }
    async ModifyDashboardSubscribe(req, cb) {
        return this.request("ModifyDashboardSubscribe", req, cb);
    }
    async ModifyDashboard(req, cb) {
        return this.request("ModifyDashboard", req, cb);
    }
    async CreateNetworkApplication(req, cb) {
        return this.request("CreateNetworkApplication", req, cb);
    }
    async DeleteWebCallback(req, cb) {
        return this.request("DeleteWebCallback", req, cb);
    }
    async PreviewKafkaRecharge(req, cb) {
        return this.request("PreviewKafkaRecharge", req, cb);
    }
    async CloseKafkaConsumer(req, cb) {
        return this.request("CloseKafkaConsumer", req, cb);
    }
    async ModifyConfigExtra(req, cb) {
        return this.request("ModifyConfigExtra", req, cb);
    }
    async ModifyAlarmShield(req, cb) {
        return this.request("ModifyAlarmShield", req, cb);
    }
    async SearchDashboardSubscribe(req, cb) {
        return this.request("SearchDashboardSubscribe", req, cb);
    }
    async CreateConfigExtra(req, cb) {
        return this.request("CreateConfigExtra", req, cb);
    }
    async CreateHostMetricConfig(req, cb) {
        return this.request("CreateHostMetricConfig", req, cb);
    }
    async DescribeRebuildIndexTasks(req, cb) {
        return this.request("DescribeRebuildIndexTasks", req, cb);
    }
    async DescribeAlarmShields(req, cb) {
        return this.request("DescribeAlarmShields", req, cb);
    }
    async ModifyScheduledSql(req, cb) {
        return this.request("ModifyScheduledSql", req, cb);
    }
    async CreateMachineGroup(req, cb) {
        return this.request("CreateMachineGroup", req, cb);
    }
    async DeleteMachineGroupInfo(req, cb) {
        return this.request("DeleteMachineGroupInfo", req, cb);
    }
    async DescribeConsoleSharingList(req, cb) {
        return this.request("DescribeConsoleSharingList", req, cb);
    }
    async CommitConsumerOffsets(req, cb) {
        return this.request("CommitConsumerOffsets", req, cb);
    }
    async CreateConsoleSharing(req, cb) {
        return this.request("CreateConsoleSharing", req, cb);
    }
    async DescribeLogsets(req, cb) {
        return this.request("DescribeLogsets", req, cb);
    }
    async CreateDashboardSubscribe(req, cb) {
        return this.request("CreateDashboardSubscribe", req, cb);
    }
    async DescribeTopicBaseMetricConfigs(req, cb) {
        return this.request("DescribeTopicBaseMetricConfigs", req, cb);
    }
    async ModifyNoticeContent(req, cb) {
        return this.request("ModifyNoticeContent", req, cb);
    }
    async SearchCosRechargeInfo(req, cb) {
        return this.request("SearchCosRechargeInfo", req, cb);
    }
    async DescribeIndex(req, cb) {
        return this.request("DescribeIndex", req, cb);
    }
    async DescribeWebCallbacks(req, cb) {
        return this.request("DescribeWebCallbacks", req, cb);
    }
    async DescribeSplunkDelivers(req, cb) {
        return this.request("DescribeSplunkDelivers", req, cb);
    }
    async CreateAlarmNotice(req, cb) {
        return this.request("CreateAlarmNotice", req, cb);
    }
    async ModifyConsumerGroup(req, cb) {
        return this.request("ModifyConsumerGroup", req, cb);
    }
    async DeleteDashboardSubscribe(req, cb) {
        return this.request("DeleteDashboardSubscribe", req, cb);
    }
    async CreateRebuildIndexTask(req, cb) {
        return this.request("CreateRebuildIndexTask", req, cb);
    }
    async DeleteMetricConfig(req, cb) {
        return this.request("DeleteMetricConfig", req, cb);
    }
    async CreateSplunkDeliver(req, cb) {
        return this.request("CreateSplunkDeliver", req, cb);
    }
    async CreateDataTransform(req, cb) {
        return this.request("CreateDataTransform", req, cb);
    }
    async AddMachineGroupInfo(req, cb) {
        return this.request("AddMachineGroupInfo", req, cb);
    }
    async DeleteNoticeContent(req, cb) {
        return this.request("DeleteNoticeContent", req, cb);
    }
    async DeleteConsoleSharing(req, cb) {
        return this.request("DeleteConsoleSharing", req, cb);
    }
    async DescribeAlarmNotices(req, cb) {
        return this.request("DescribeAlarmNotices", req, cb);
    }
    async DescribePartitions(req, cb) {
        return this.request("DescribePartitions", req, cb);
    }
    async DeleteConfigExtra(req, cb) {
        return this.request("DeleteConfigExtra", req, cb);
    }
    async CheckFunction(req, cb) {
        return this.request("CheckFunction", req, cb);
    }
    async SearchLog(req, cb) {
        return this.request("SearchLog", req, cb);
    }
    async DeleteShipper(req, cb) {
        return this.request("DeleteShipper", req, cb);
    }
    async CreateWebCallback(req, cb) {
        return this.request("CreateWebCallback", req, cb);
    }
    async QueryRangeMetric(req, cb) {
        return this.request("QueryRangeMetric", req, cb);
    }
    async DescribeConfigMachineGroups(req, cb) {
        return this.request("DescribeConfigMachineGroups", req, cb);
    }
    async DeleteExport(req, cb) {
        return this.request("DeleteExport", req, cb);
    }
    async SplitPartition(req, cb) {
        return this.request("SplitPartition", req, cb);
    }
    async DescribeDataTransformInfo(req, cb) {
        return this.request("DescribeDataTransformInfo", req, cb);
    }
    async DescribeMachineGroups(req, cb) {
        return this.request("DescribeMachineGroups", req, cb);
    }
    async CreateConsumer(req, cb) {
        return this.request("CreateConsumer", req, cb);
    }
    async DescribeMetricSubscribePreview(req, cb) {
        return this.request("DescribeMetricSubscribePreview", req, cb);
    }
    async ModifyTopic(req, cb) {
        return this.request("ModifyTopic", req, cb);
    }
    async DescribeDashboardSubscribes(req, cb) {
        return this.request("DescribeDashboardSubscribes", req, cb);
    }
    async DescribeShipperTasks(req, cb) {
        return this.request("DescribeShipperTasks", req, cb);
    }
    async ModifyWebCallback(req, cb) {
        return this.request("ModifyWebCallback", req, cb);
    }
    async ModifyMachineGroup(req, cb) {
        return this.request("ModifyMachineGroup", req, cb);
    }
    async DescribeKafkaConsumerTopics(req, cb) {
        return this.request("DescribeKafkaConsumerTopics", req, cb);
    }
    async DescribeConsumerGroups(req, cb) {
        return this.request("DescribeConsumerGroups", req, cb);
    }
    async DescribeScheduledSqlInfo(req, cb) {
        return this.request("DescribeScheduledSqlInfo", req, cb);
    }
    async DeleteConfig(req, cb) {
        return this.request("DeleteConfig", req, cb);
    }
    async DescribeConfigExtras(req, cb) {
        return this.request("DescribeConfigExtras", req, cb);
    }
    async CheckRechargeKafkaServer(req, cb) {
        return this.request("CheckRechargeKafkaServer", req, cb);
    }
    async ModifyAlarm(req, cb) {
        return this.request("ModifyAlarm", req, cb);
    }
    async DescribeShippers(req, cb) {
        return this.request("DescribeShippers", req, cb);
    }
    async ModifyKafkaConsumerGroupOffset(req, cb) {
        return this.request("ModifyKafkaConsumerGroupOffset", req, cb);
    }
    async ModifyConsoleSharing(req, cb) {
        return this.request("ModifyConsoleSharing", req, cb);
    }
    async DescribeMetricSubscribes(req, cb) {
        return this.request("DescribeMetricSubscribes", req, cb);
    }
    async GetMetricLabelValues(req, cb) {
        return this.request("GetMetricLabelValues", req, cb);
    }
    async DescribeConsumers(req, cb) {
        return this.request("DescribeConsumers", req, cb);
    }
    async DeleteNetworkApplication(req, cb) {
        return this.request("DeleteNetworkApplication", req, cb);
    }
    async DescribeKafkaConsumerGroupDetail(req, cb) {
        return this.request("DescribeKafkaConsumerGroupDetail", req, cb);
    }
    async DescribeDashboards(req, cb) {
        return this.request("DescribeDashboards", req, cb);
    }
    async DeleteAlarmNotice(req, cb) {
        return this.request("DeleteAlarmNotice", req, cb);
    }
    async ModifyCosRecharge(req, cb) {
        return this.request("ModifyCosRecharge", req, cb);
    }
    async ModifyConfig(req, cb) {
        return this.request("ModifyConfig", req, cb);
    }
    async UploadLog(req, cb) {
        return this.request("UploadLog", req, cb);
    }
    async CreateScheduledSql(req, cb) {
        return this.request("CreateScheduledSql", req, cb);
    }
    async ModifyLogset(req, cb) {
        return this.request("ModifyLogset", req, cb);
    }
    async ModifyConsumer(req, cb) {
        return this.request("ModifyConsumer", req, cb);
    }
    async DescribeMachineGroupConfigs(req, cb) {
        return this.request("DescribeMachineGroupConfigs", req, cb);
    }
    async ModifyDataTransform(req, cb) {
        return this.request("ModifyDataTransform", req, cb);
    }
    async DeleteAlarm(req, cb) {
        return this.request("DeleteAlarm", req, cb);
    }
    async DescribeLogContext(req, cb) {
        return this.request("DescribeLogContext", req, cb);
    }
    async DescribeConsumer(req, cb) {
        return this.request("DescribeConsumer", req, cb);
    }
    async CreateMetricConfig(req, cb) {
        return this.request("CreateMetricConfig", req, cb);
    }
    async DescribeNoticeContents(req, cb) {
        return this.request("DescribeNoticeContents", req, cb);
    }
    async DeleteIndex(req, cb) {
        return this.request("DeleteIndex", req, cb);
    }
    async ModifyMetricConfig(req, cb) {
        return this.request("ModifyMetricConfig", req, cb);
    }
    async DescribeMachines(req, cb) {
        return this.request("DescribeMachines", req, cb);
    }
    async RetryShipperTask(req, cb) {
        return this.request("RetryShipperTask", req, cb);
    }
    async DescribeKafkaConsumer(req, cb) {
        return this.request("DescribeKafkaConsumer", req, cb);
    }
    async CreateAlarm(req, cb) {
        return this.request("CreateAlarm", req, cb);
    }
    async DeleteTopic(req, cb) {
        return this.request("DeleteTopic", req, cb);
    }
}
