// tencentcloud-sdk-nodejs/es/services/ai3d/v20250513/ai3d_client.js
import { AbstractClient } from "../../../common/abstract_client";
export class Client extends AbstractClient {
    constructor(clientConfig) {
        super("ai3d.tencentcloudapi.com", "2025-05-13", clientConfig);
    }
    async SubmitTextureTo3DJob(req, cb) {
        return this.request("SubmitTextureTo3DJob", req, cb);
    }
    async SubmitReduceFaceJob(req, cb) {
        return this.request("SubmitReduceFaceJob", req, cb);
    }
    async SubmitHunyuanTo3DProJob(req, cb) {
        return this.request("SubmitHunyuanTo3DProJob", req, cb);
    }
    async SubmitHunyuanTo3DRapidJob(req, cb) {
        return this.request("SubmitHunyuanTo3DRapidJob", req, cb);
    }
    async SubmitProfileTo3DJob(req, cb) {
        return this.request("SubmitProfileTo3DJob", req, cb);
    }
    async SubmitHunyuanTo3DUVJob(req, cb) {
        return this.request("SubmitHunyuanTo3DUVJob", req, cb);
    }
    async SubmitHunyuan3DPartJob(req, cb) {
        return this.request("SubmitHunyuan3DPartJob", req, cb);
    }
    async DescribeReduceFaceJob(req, cb) {
        return this.request("DescribeReduceFaceJob", req, cb);
    }
    async QueryHunyuanTo3DRapidJob(req, cb) {
        return this.request("QueryHunyuanTo3DRapidJob", req, cb);
    }
    async QueryHunyuan3DPartJob(req, cb) {
        return this.request("QueryHunyuan3DPartJob", req, cb);
    }
    async DescribeTextureTo3DJob(req, cb) {
        return this.request("DescribeTextureTo3DJob", req, cb);
    }
    async DescribeProfileTo3DJob(req, cb) {
        return this.request("DescribeProfileTo3DJob", req, cb);
    }
    async DescribeHunyuanTo3DUVJob(req, cb) {
        return this.request("DescribeHunyuanTo3DUVJob", req, cb);
    }
    async QueryHunyuanTo3DProJob(req, cb) {
        return this.request("QueryHunyuanTo3DProJob", req, cb);
    }
    async Convert3DFormat(req, cb) {
        return this.request("Convert3DFormat", req, cb);
    }
}
