// tencentcloud-sdk-nodejs/tencentcloud/services/tdmysql/v20211122/tdmysql_client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
/* eslint-disable @typescript-eslint/no-unused-vars */
/*
 * Copyright (c) 2018 Tencent. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
const abstract_client_1 = require("../../../common/abstract_client");
/**
 * tdmysql client
 * @class
 */
class Client extends abstract_client_1.AbstractClient {
    constructor(clientConfig) {
        super("tdmysql.tencentcloudapi.com", "2021-11-22", clientConfig);
    }
    /**
     * 本接口（DescribeDatabaseObjects）用于查询云数据库实例的数据库中的对象列表，包含表、存储过程、视图和函数。
     */
    async DescribeDatabaseObjects(req, cb) {
        return this.request("DescribeDatabaseObjects", req, cb);
    }
    /**
     * 本接口（DescribeDBParameters）用于获取实例的当前参数设置。
     */
    async DescribeDBParameters(req, cb) {
        return this.request("DescribeDBParameters", req, cb);
    }
    /**
     * 本接口（DescribeDBSecurityGroups）用于查询实例安全组信息
     */
    async DescribeDBSecurityGroups(req, cb) {
        return this.request("DescribeDBSecurityGroups", req, cb);
    }
    /**
     * 本接口（CancelIsolateDBInstances）提供批量解除隔离实例功能
     */
    async CancelIsolateDBInstances(req, cb) {
        return this.request("CancelIsolateDBInstances", req, cb);
    }
    /**
     * 可恢复时间查询
     */
    async DescribeDBSAvailableRecoveryTime(req, cb) {
        return this.request("DescribeDBSAvailableRecoveryTime", req, cb);
    }
    /**
     * 修改备份集备注
     */
    async ModifyDBSBackupSetComment(req, cb) {
        return this.request("ModifyDBSBackupSetComment", req, cb);
    }
    /**
     * 删除实例备份集
     */
    async DeleteDBSBackupSets(req, cb) {
        return this.request("DeleteDBSBackupSets", req, cb);
    }
    /**
     * 修改实例备份策略
     */
    async ModifyDBSBackupPolicy(req, cb) {
        return this.request("ModifyDBSBackupPolicy", req, cb);
    }
    /**
     * 本接口（DestroyInstances）提供批量销毁实例功能
     */
    async DestroyInstances(req, cb) {
        return this.request("DestroyInstances", req, cb);
    }
    /**
     * 本接口（ModifyAutoRenewFlag）用于修改自动续费标志
     */
    async ModifyAutoRenewFlag(req, cb) {
        return this.request("ModifyAutoRenewFlag", req, cb);
    }
    /**
     * 创建实例备份集
     */
    async CreateDBSBackup(req, cb) {
        return this.request("CreateDBSBackup", req, cb);
    }
    /**
     * 本接口（ModifyInstanceName）提供修改实例名称功能
     */
    async ModifyInstanceName(req, cb) {
        return this.request("ModifyInstanceName", req, cb);
    }
    /**
     * 本接口（ModifyDBParameters）用于修改实例参数。
     */
    async ModifyDBParameters(req, cb) {
        return this.request("ModifyDBParameters", req, cb);
    }
    /**
     * 查询实例克隆列表
     */
    async DescribeDBSCloneInstances(req, cb) {
        return this.request("DescribeDBSCloneInstances", req, cb);
    }
    /**
     * 本接口（IsolateDBInstance）提供批量隔离实例功能
     */
    async IsolateDBInstance(req, cb) {
        return this.request("IsolateDBInstance", req, cb);
    }
    /**
     * 本接口（ModifyDBInstanceSecurityGroups）用于修改云数据库安全组
     */
    async ModifyDBInstanceSecurityGroups(req, cb) {
        return this.request("ModifyDBInstanceSecurityGroups", req, cb);
    }
    /**
     * 本接口（DescribeFlow）用于查询异步任务流程状态
     */
    async DescribeFlow(req, cb) {
        return this.request("DescribeFlow", req, cb);
    }
}
exports.Client = Client;
