// tencentcloud-sdk-nodejs/es/services/tcbr/v20220217/tcbr_client.js
import { AbstractClient } from "../../../common/abstract_client";
export class Client extends AbstractClient {
    constructor(clientConfig) {
        super("tcbr.tencentcloudapi.com", "2022-02-17", clientConfig);
    }
    async DeleteCloudRunVersions(req, cb) {
        return this.request("DeleteCloudRunVersions", req, cb);
    }
    async CreateCloudRunServer(req, cb) {
        return this.request("CreateCloudRunServer", req, cb);
    }
    async DescribeServerManageTask(req, cb) {
        return this.request("DescribeServerManageTask", req, cb);
    }
    async DescribeCloudRunPodList(req, cb) {
        return this.request("DescribeCloudRunPodList", req, cb);
    }
    async DescribeCloudRunServers(req, cb) {
        return this.request("DescribeCloudRunServers", req, cb);
    }
    async DescribeCloudRunProcessLog(req, cb) {
        return this.request("DescribeCloudRunProcessLog", req, cb);
    }
    async DescribeReleaseOrder(req, cb) {
        return this.request("DescribeReleaseOrder", req, cb);
    }
    async OperateServerManage(req, cb) {
        return this.request("OperateServerManage", req, cb);
    }
    async ReleaseGray(req, cb) {
        return this.request("ReleaseGray", req, cb);
    }
    async DeleteCloudRunServer(req, cb) {
        return this.request("DeleteCloudRunServer", req, cb);
    }
    async DescribeVersionDetail(req, cb) {
        return this.request("DescribeVersionDetail", req, cb);
    }
    async CreateCloudRunEnv(req, cb) {
        return this.request("CreateCloudRunEnv", req, cb);
    }
    async SubmitServerRollback(req, cb) {
        return this.request("SubmitServerRollback", req, cb);
    }
    async DescribeCloudRunEnvs(req, cb) {
        return this.request("DescribeCloudRunEnvs", req, cb);
    }
    async SearchClsLog(req, cb) {
        return this.request("SearchClsLog", req, cb);
    }
    async DescribeCloudRunDeployRecord(req, cb) {
        return this.request("DescribeCloudRunDeployRecord", req, cb);
    }
    async DescribeEnvBaseInfo(req, cb) {
        return this.request("DescribeEnvBaseInfo", req, cb);
    }
    async UpdateCloudRunServer(req, cb) {
        return this.request("UpdateCloudRunServer", req, cb);
    }
    async DescribeCloudRunServerDetail(req, cb) {
        return this.request("DescribeCloudRunServerDetail", req, cb);
    }
}
