// tencentcloud-sdk-nodejs/tencentcloud/services/ai3d/v20250513/ai3d_client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
/* eslint-disable @typescript-eslint/no-unused-vars */
/*
 * Copyright (c) 2018 Tencent. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
const abstract_client_1 = require("../../../common/abstract_client");
/**
 * ai3d client
 * @class
 */
class Client extends abstract_client_1.AbstractClient {
    constructor(clientConfig) {
        super("ai3d.tencentcloudapi.com", "2025-05-13", clientConfig);
    }
    /**
     * 混元生3D接口，输入单几何模型和参考图或文字描述后，可生成对应的纹理贴图。
默认提供1个并发，代表最多能同时处理1个已提交的任务，上一个任务处理完毕后，才能开始处理下一个任务。
     */
    async SubmitTextureTo3DJob(req, cb) {
        return this.request("SubmitTextureTo3DJob", req, cb);
    }
    /**
     * 混元生3D接口，采用 Polygon 1.5模型，输入3D 高模后，可生成布线规整，较低面数的3D 模型。
默认提供1个并发，代表最多能同时处理1个已提交的任务，上一个任务处理完毕后，才能开始处理下一个任务。
     */
    async SubmitReduceFaceJob(req, cb) {
        return this.request("SubmitReduceFaceJob", req, cb);
    }
    /**
     * 混元生3D接口，基于混元大模型，根据输入的文本描述/图片智能生成3D。
默认提供3个并发，代表最多能同时处理3个已提交的任务，上一个任务处理完毕后，才能开始处理下一个任务。
     */
    async SubmitHunyuanTo3DProJob(req, cb) {
        return this.request("SubmitHunyuanTo3DProJob", req, cb);
    }
    /**
     * 混元生3D接口，基于混元大模型，根据输入的文本描述/图片智能生成3D。
默认提供1个并发，代表最多能同时处理1个已提交的任务，上一个任务处理完毕后，才能开始处理下一个任务。
     */
    async SubmitHunyuanTo3DRapidJob(req, cb) {
        return this.request("SubmitHunyuanTo3DRapidJob", req, cb);
    }
    /**
     * 混元生3D接口，基于混元大模型，根据输入的文本描述/图片智能生成3D。
默认提供1个并发，代表最多能同时处理1个已提交的任务，上一个任务处理完毕后，才能开始处理下一个任务。
     */
    async SubmitProfileTo3DJob(req, cb) {
        return this.request("SubmitProfileTo3DJob", req, cb);
    }
    /**
     * 输入模型后，可根据模型纹理进行UV展开，输出对应UV贴图。
     */
    async SubmitHunyuanTo3DUVJob(req, cb) {
        return this.request("SubmitHunyuanTo3DUVJob", req, cb);
    }
    /**
     * 输入3D模型文件后，根据模型结构自动进行组件识别生成。
     */
    async SubmitHunyuan3DPartJob(req, cb) {
        return this.request("SubmitHunyuan3DPartJob", req, cb);
    }
    /**
     * 混元生3D接口，采用 Polygon 1.5模型，输入3D 高模后，可生成布线规整，较低面数的3D 模型。
默认提供1个并发，代表最多能同时处理1个已提交的任务，上一个任务处理完毕后，才能开始处理下一个任务。
     */
    async DescribeReduceFaceJob(req, cb) {
        return this.request("DescribeReduceFaceJob", req, cb);
    }
    /**
     * 混元生3D接口，基于混元大模型，根据输入的文本描述/图片智能生成3D。
默认提供1个并发，代表最多能同时处理1个已提交的任务，上一个任务处理完毕后，才能开始处理下一个任务。
     */
    async QueryHunyuanTo3DRapidJob(req, cb) {
        return this.request("QueryHunyuanTo3DRapidJob", req, cb);
    }
    /**
     * 查询组件生成任务。
     */
    async QueryHunyuan3DPartJob(req, cb) {
        return this.request("QueryHunyuan3DPartJob", req, cb);
    }
    /**
     * 混元生3D接口，输入单几何模型和参考图或文字描述后，可生成对应的纹理贴图。
默认提供1个并发，代表最多能同时处理1个已提交的任务，上一个任务处理完毕后，才能开始处理下一个任务。
     */
    async DescribeTextureTo3DJob(req, cb) {
        return this.request("DescribeTextureTo3DJob", req, cb);
    }
    /**
     * 混元生3D接口，基于混元大模型，根据输入的文本描述/图片智能生成3D。
默认提供1个并发，代表最多能同时处理1个已提交的任务，上一个任务处理完毕后，才能开始处理下一个任务。
     */
    async DescribeProfileTo3DJob(req, cb) {
        return this.request("DescribeProfileTo3DJob", req, cb);
    }
    /**
     * 查询组件拆分任务。
     */
    async DescribeHunyuanTo3DUVJob(req, cb) {
        return this.request("DescribeHunyuanTo3DUVJob", req, cb);
    }
    /**
     * 混元生3D接口，基于混元大模型，根据输入的文本描述/图片智能生成3D。
默认提供3个并发，代表最多能同时处理3个已提交的任务，上一个任务处理完毕后，才能开始处理下一个任务。
     */
    async QueryHunyuanTo3DProJob(req, cb) {
        return this.request("QueryHunyuanTo3DProJob", req, cb);
    }
    /**
     * 输入3D模型文件后，可进行3D模型文件格式转换。
     */
    async Convert3DFormat(req, cb) {
        return this.request("Convert3DFormat", req, cb);
    }
}
exports.Client = Client;
