// tencentcloud-sdk-nodejs/es/services/apm/v20210622/apm_client.js
import { AbstractClient } from "../../../common/abstract_client";
export class Client extends AbstractClient {
    constructor(clientConfig) {
        super("apm.tencentcloudapi.com", "2021-06-22", clientConfig);
    }
    async ModifyApmInstance(req, cb) {
        return this.request("ModifyApmInstance", req, cb);
    }
    async DescribeApmVulnerabilityDetail(req, cb) {
        return this.request("DescribeApmVulnerabilityDetail", req, cb);
    }
    async DescribeGeneralSpanList(req, cb) {
        return this.request("DescribeGeneralSpanList", req, cb);
    }
    async DescribeGeneralOTSpanList(req, cb) {
        return this.request("DescribeGeneralOTSpanList", req, cb);
    }
    async CreateApmInstance(req, cb) {
        return this.request("CreateApmInstance", req, cb);
    }
    async DescribeTopologyNew(req, cb) {
        return this.request("DescribeTopologyNew", req, cb);
    }
    async DescribeGeneralApmApplicationConfig(req, cb) {
        return this.request("DescribeGeneralApmApplicationConfig", req, cb);
    }
    async ModifyApmAssociation(req, cb) {
        return this.request("ModifyApmAssociation", req, cb);
    }
    async ModifyApmSampleConfig(req, cb) {
        return this.request("ModifyApmSampleConfig", req, cb);
    }
    async DescribeApmAgent(req, cb) {
        return this.request("DescribeApmAgent", req, cb);
    }
    async DescribeGeneralMetricData(req, cb) {
        return this.request("DescribeGeneralMetricData", req, cb);
    }
    async CreateApmPrometheusRule(req, cb) {
        return this.request("CreateApmPrometheusRule", req, cb);
    }
    async TerminateApmInstance(req, cb) {
        return this.request("TerminateApmInstance", req, cb);
    }
    async DescribeApmAssociation(req, cb) {
        return this.request("DescribeApmAssociation", req, cb);
    }
    async DescribeApmServiceMetric(req, cb) {
        return this.request("DescribeApmServiceMetric", req, cb);
    }
    async ModifyApmPrometheusRule(req, cb) {
        return this.request("ModifyApmPrometheusRule", req, cb);
    }
    async DescribeApmSQLInjectionDetail(req, cb) {
        return this.request("DescribeApmSQLInjectionDetail", req, cb);
    }
    async CreateApmSampleConfig(req, cb) {
        return this.request("CreateApmSampleConfig", req, cb);
    }
    async DescribeApmSampleConfig(req, cb) {
        return this.request("DescribeApmSampleConfig", req, cb);
    }
    async DescribeTagValues(req, cb) {
        return this.request("DescribeTagValues", req, cb);
    }
    async DescribeApmPrometheusRule(req, cb) {
        return this.request("DescribeApmPrometheusRule", req, cb);
    }
    async DescribeApmInstances(req, cb) {
        return this.request("DescribeApmInstances", req, cb);
    }
    async DeleteApmSampleConfig(req, cb) {
        return this.request("DeleteApmSampleConfig", req, cb);
    }
    async CreateProfileTask(req, cb) {
        return this.request("CreateProfileTask", req, cb);
    }
    async DescribeApmAllVulCount(req, cb) {
        return this.request("DescribeApmAllVulCount", req, cb);
    }
    async DescribeApmApplicationConfig(req, cb) {
        return this.request("DescribeApmApplicationConfig", req, cb);
    }
    async ModifyGeneralApmApplicationConfig(req, cb) {
        return this.request("ModifyGeneralApmApplicationConfig", req, cb);
    }
    async DescribeMetricRecords(req, cb) {
        return this.request("DescribeMetricRecords", req, cb);
    }
    async ModifyApmApplicationConfig(req, cb) {
        return this.request("ModifyApmApplicationConfig", req, cb);
    }
    async DescribeOPRAllVulCount(req, cb) {
        return this.request("DescribeOPRAllVulCount", req, cb);
    }
    async DescribeServiceOverview(req, cb) {
        return this.request("DescribeServiceOverview", req, cb);
    }
    async DescribeApmVulnerabilityCount(req, cb) {
        return this.request("DescribeApmVulnerabilityCount", req, cb);
    }
}
