// tencentcloud-sdk-nodejs/es/services/vclm/v20240523/vclm_client.js
import { AbstractClient } from "../../../common/abstract_client";
export class Client extends AbstractClient {
    constructor(clientConfig) {
        super("vclm.tencentcloudapi.com", "2024-05-23", clientConfig);
    }
    async SubmitHunyuanToVideoJob(req, cb) {
        return this.request("SubmitHunyuanToVideoJob", req, cb);
    }
    async DescribeVideoEditJob(req, cb) {
        return this.request("DescribeVideoEditJob", req, cb);
    }
    async SubmitPortraitSingJob(req, cb) {
        return this.request("SubmitPortraitSingJob", req, cb);
    }
    async SubmitVideoFaceFusionJob(req, cb) {
        return this.request("SubmitVideoFaceFusionJob", req, cb);
    }
    async CheckAnimateImageJob(req, cb) {
        return this.request("CheckAnimateImageJob", req, cb);
    }
    async DescribeImageToVideoGeneralJob(req, cb) {
        return this.request("DescribeImageToVideoGeneralJob", req, cb);
    }
    async DescribeTemplateToVideoJob(req, cb) {
        return this.request("DescribeTemplateToVideoJob", req, cb);
    }
    async SubmitVideoStylizationJob(req, cb) {
        return this.request("SubmitVideoStylizationJob", req, cb);
    }
    async DescribePortraitSingJob(req, cb) {
        return this.request("DescribePortraitSingJob", req, cb);
    }
    async DescribeAigcVideoJob(req, cb) {
        return this.request("DescribeAigcVideoJob", req, cb);
    }
    async SubmitVideoEditJob(req, cb) {
        return this.request("SubmitVideoEditJob", req, cb);
    }
    async DescribeVideoVoiceJob(req, cb) {
        return this.request("DescribeVideoVoiceJob", req, cb);
    }
    async SubmitAigcVideoJob(req, cb) {
        return this.request("SubmitAigcVideoJob", req, cb);
    }
    async SubmitVideoVoiceJob(req, cb) {
        return this.request("SubmitVideoVoiceJob", req, cb);
    }
    async SubmitHumanActorJob(req, cb) {
        return this.request("SubmitHumanActorJob", req, cb);
    }
    async DescribeHunyuanToVideoJob(req, cb) {
        return this.request("DescribeHunyuanToVideoJob", req, cb);
    }
    async DescribeImageAnimateJob(req, cb) {
        return this.request("DescribeImageAnimateJob", req, cb);
    }
    async SubmitImageAnimateJob(req, cb) {
        return this.request("SubmitImageAnimateJob", req, cb);
    }
    async DescribeVideoStylizationJob(req, cb) {
        return this.request("DescribeVideoStylizationJob", req, cb);
    }
    async DescribeHumanActorJob(req, cb) {
        return this.request("DescribeHumanActorJob", req, cb);
    }
    async DescribeVideoFaceFusionJob(req, cb) {
        return this.request("DescribeVideoFaceFusionJob", req, cb);
    }
    async SubmitTemplateToVideoJob(req, cb) {
        return this.request("SubmitTemplateToVideoJob", req, cb);
    }
    async SubmitImageToVideoGeneralJob(req, cb) {
        return this.request("SubmitImageToVideoGeneralJob", req, cb);
    }
}
